### PR TITLE
MGTwoLevelTransfer: refine DG check

### PR DESCRIPTION
**Pull request by peterrum**
There are cases where `fine_element_is_continuous==false` (DG) is not enough to skip communication. The degrees of freedoms might belong to a ghost cell.

@kronbichler Do you agree. This was triggered today in ExaDG.

FYI @richardschu

### CODE REVIEW

File: include/deal.II/multigrid/mg_transfer_global_coarsening.h [hunk 1]
```diff
@@ -389,9 +395,9 @@ class MGTwoLevelTransferBase<LinearAlgebra::distributed::Vector<Number>>
   mutable LinearAlgebra::distributed::Vector<Number> vec_fine;
 
   /**
-   * Internal vector on that the actual prolongation/restriction is performed.
+   * Bool indicating whether fine vector has relevant ghost values.
    */
-  mutable LinearAlgebra::distributed::Vector<Number> vec_coarse;
+  bool vec_fine_needs_ghost_update = true;
```
Comment (kronbichler) [inline include/deal.II/multigrid/mg_transfer_global_coarsening.h:400]: I think we do not initialize variables at the place of declaration.

**Review by kronbichler — APPROVED** [review]
Yes, indeed this check makes sense because it covers the identity we actually rely on. It is surprising that some cells get ghosted in case of a pure `p` multigrid setup (is this the use case), but either way this should be uncritical.

**Comment by kronbichler** [comment]
Is there a way to test this?

**Comment by peterrum** [comment]
> Is there a way to test this?

I think so. The problem occurred during pure h-multigrid (with ghost cells due to DG). I'll write a simplified test tomorrow.

**Comment by peterrum** [comment]
@kronbichler I have pushed a simple test: 2 cells and 2 processes.

**Comment by kronbichler** [comment]
Great, that's a good case.

**Review by kronbichler — APPROVED** [review]

**Pull request merged**